service_provider.go

```diff
@@ -261,24 +261,30 @@ func (req *AuthnRequest) Redirect(relayState string, sp *ServiceProvider) (*url.
 	rv, _ := url.Parse(req.Destination)
 	// We can't depend on Query().set() as order matters for signing
 	query := rv.RawQuery
+
 	if len(query) > 0 {
 		query += "&SAMLRequest=" + url.QueryEscape(string(w.Bytes()))
 	} else {
 		query += "SAMLRequest=" + url.QueryEscape(string(w.Bytes()))
 	}
 
+	queryToBeSigned := "SAMLRequest=" + url.QueryEscape(string(w.Bytes()))
```

Comment:
could you not have simply removed lines 265-267? and is there a reason to preserve the `query` variable which is no longer used?

Reply:
> could you not have simply removed lines 265-267? and is there a reason to preserve the `query` variable which is no longer used?

We are using query variable to send the original request line#291 and 294

Reply:
ah, I see it right below the diff window. Thanks for clarifying!